content/blog/prototyping/index.md

```diff
@@ -0,0 +1,732 @@
+
+## Why You Should Prototype 
+
+As much as we like to come up with the perfect solution from the start, it rarely works that way.
+Programming is an iterative process.
+
+In my experience, prototyping solutions helps tremendously in finding the best approach before committing to a design.
+
+This iterative process isn't just useful for writing games – it's equally valuable when crafting a CLI tool where you need to figure out the command line interface, or when designing a library where you need to nail down the API.
+
+**People prototype because they want to explore the design space.
+When done right, this process leads to more idiomatic code.**
+
+## Why People Believe Rust Is Not Good For Prototyping
+
+The common narrative goes like this:
+
+> When you start writing a program, you don't know what you want and you change your mind pretty often.
+> Rust pushes back when you change your mind because the type system is very strict.
+> On top of that, getting your idea to compile takes longer than in other languages, so the feedback loop is slower.
+
+That's why people believe that Rust isn't a good fit for prototyping.
+
+I've found that developers from other languages and Rust beginners often share this preconception about Rust.
+These developers stumble over the strict type system and the borrow checker while trying to sketch out a solution.
+They believe that with Rust you're either at 0% or 100% done (everything works and has no undefined behavior) and there's nothing in between.
+
+Here are some typical misbeliefs: 
+
+1. Rust always requires you to handle errors.
+2. Memory safety gets in the way of prototyping.
+3. Ownership and borrowing take out the fun of prototyping.
+4. You have to get all the details right from the beginning. 
+
+These are all [common misconceptions](https://medium.com/@victor.ronin/love-hate-relationship-with-rust-language-part-2-c36f57d5485d) and they are not true. 
+
+It turns out you can avoid all of these pitfalls and still get a lot of value from prototyping in Rust. 
+The interaction between types is the central part of the prototyping stage, and that's where Rust shines compared to other languages.
+
+## Problems with Prototyping in Other Languages
+
+If you're happy with a scripting language like Python, why bother with Rust?
+
+That's a fair question.
+After all, Python is known for its quick feedback loop and dynamic type system, and you can always rewrite the code in Rust later.
+
+Yes, Python is a great choice for prototyping.
+But I've been a Python developer for a long time, and I find that I very quickly grow out of the "prototype" phase – which is when the language falls apart for me.
+
+One thing I found particularly challenging in Python was hardening my prototype into a robust, production-ready codebase.
```

Comment:
This is indeed the case, but the article is about prototyping, not day 2, not sure you want this paragraph and I'd maybe even adjust the one below

Reply:
True. I'll clean that up.

Reply:
Modified.